### PR TITLE
FE tuned up with latest versions of dependencies

### PR DESCRIPTION
**Pull request by ug02fast**
Tuned up the FE app with latest versions for all dependencies including some major changes like: `react-scripts v5, react-router-dom v6 and Cypress v9`

`ethcall` upgrade was problematic as it was erroring in regards to an invalid blockTag. I will look into upgrading that package at another time.

### CODE REVIEW

File: src/components/TopMenu.tsx [hunk 1]
```diff
@@ -111,7 +111,7 @@ function TopMenu(): ReactElement {
             <NetworkDisplay onClick={handleSettingMenu} />
             <IconButton
               onClick={handleSettingMenu}
-              data-testId="settingsMenuBtn"
+              data-testid="settingsMenuBtn"
```
Comment (ug02fast) [inline src/components/TopMenu.tsx:114]: dangling bug

File: .eslintrc [hunk 2]
```diff
@@ -30,6 +29,13 @@
   "rules": {
     "@typescript-eslint/no-unused-vars": ["error"],
     "@typescript-eslint/ban-ts-comment": "off",
+    "@typescript-eslint/no-misused-promises": [
```
Comment (ug02fast) [inline .eslintrc:32]: got stuck figuring this out for a long time. can't figure it out

File: package.json [hunk 3]
```diff
@@ -138,17 +133,5 @@
...
     "*.{ts,tsx}": "eslint --cache --fix"
-  },
-  "eslintConfig": {
-    "overrides": [
-      {
-        "files": [
-          "**/*.stories.*"
-        ],
-        "rules": {
-          "import/no-anonymous-default-export": "off"
```
Comment (ug02fast) [inline package.json:149]: unused

Comment (nkrulikovsky) [reply package.json:149]: Good remove!

File: src/pages/App.tsx [hunk 4]
```diff
@@ -74,31 +74,27 @@ export default function App(): ReactElement {
...
+                      path={`pools/${route}/deposit`}
+                      element={<Deposit poolName={name} />}
+                      key={`${name}-deposit`}
                     />
                   ))}
-                  <Redirect from="/pools/:route/:action" to="/pools" />
-                  <Route exact path="/risk" component={Risk} />
-                  <Route exact path="/vesting-claim" component={VestingClaim} />
-                </Switch>
+                  <Route path="pools/*" element={<Pools />} />
```
Comment (nkrulikovsky) [inline src/pages/App.tsx:94]: Nice update!

File: src/utils/index.ts [hunk 5]
```diff
@@ -204,25 +204,9 @@ export function getTokenSymbolForPoolType(poolType: PoolTypes): string {
...
-    ethcallProvider.multicallAddress =
-      "0xab16069d3e9e352343b2040ce7d7715c585994f9"
-  } else if (chainId === ChainId.OPTIMISM) {
-    ethcallProvider.multicallAddress =
-      "0x266557A864680A1401A3506c0eb72934BD13Bf59"
-  } else if (chainId === ChainId.FANTOM) {
-    ethcallProvider.multicallAddress =
-      "0xc04d660976c923ddba750341fe5923e47900cf24"
-  }
+
```
Comment (nkrulikovsky) [inline src/utils/index.ts:210]: Nice update!

**Review by nkrulikovsky — COMMENTED** [review]
Great job!

File: src/components/TopMenu.tsx (hunk 1 above)
Comment (nkrulikovsky) [reply src/components/TopMenu.tsx:114]: My bad typo 😊

File: package.json (hunk 3 above)
Comment (ug02fast) [reply package.json:149]: oh it must have been for the storybook stories. if it causes problems just re-add 👍

**Review by nkrulikovsky — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (nkrulikovsky) [reply package.json:149]: Yes, will do!

File: package.json [hunk 6]
```diff
@@ -78,6 +78,7 @@
     "eslint-plugin-react-hooks": "^4.3.0",
     "husky": "^7.0.4",
     "lint-staged": "^12.3.5",
+    "postcss": "^8.4.8",
```
Comment (nkrulikovsky) [inline package.json:81]: I'd love to know the advantages of this plugin in our dApp?

Comment (ug02fast) [reply package.json:81]: I believe it was a problem happening on the CI side when running the new Cypress version.

Comment (nkrulikovsky) [reply package.json:81]: `postcss` is a tool related with styles, and does it related with cypress?

Comment (ug02fast) [reply package.json:81]: It's included in react-scripts. Including it as a devDep here fixes a missing module error during CI.

**Comment by ug02fast** [comment]
image
withdraw flow passing locally

File: package.json [hunk 7]
```diff
@@ -7,95 +7,91 @@
...
-    "lodash": "^4.17.20",
-    "react": "^17.0.1",
-    "react-device-detect": "^1.14.0",
-    "react-dom": "^17.0.1",
-    "react-i18next": "^11.11.0",
-    "react-redux": "^7.2.4",
-    "react-router-dom": "^5.2.0",
-    "react-scripts": "4.0.3",
-    "react-toastify": "^8.1.1",
+    "ethcall": "3.2.0",
```
Comment (hammeiam) [inline package.json:33]: we omitting this upgrade for now?

File: src/hooks/useAddTokenToMetamask.ts [hunk 8]
```diff
@@ -33,7 +33,9 @@ export default function useAddTokenToMetamask(token: Token | undefined): {
             },
           },
         })
-        .then((success) => setSuccess(success))
+        .then((success: SetStateAction<boolean | undefined>) =>
```
Comment (hammeiam) [inline src/hooks/useAddTokenToMetamask.ts:36]: This doesn't seem correct - `success` is just a boolean and `SetStateAction` is the type for `setState` in the context `const [state, setState] = useState()`

`type SetStateAction<S> = S | ((prevState: S) => S);`

File: tsconfig.json [hunk 9]
```diff
@@ -1,11 +1,7 @@
 {
   "compilerOptions": {
-    "target": "es5",
-    "lib": [
-      "dom",
-      "dom.iterable",
-      "esnext"
-    ],
+    "target": "es6",
```
Comment (hammeiam) [inline tsconfig.json:3]: ooh does this decrease our bundle size??

**Review by hammeiam — APPROVED** [review]
One nit but looks great otherwise!

File: package.json (hunk 7 above)
Comment (ug02fast) [reply package.json:33]: Yes I'll most likely follow up on it in a separate PR

File: .eslintrc (hunk 2 above)
Comment (ug02fast) [reply .eslintrc:32]: I think I put that there out of rage. I'll remove it and see what gets called out.

File: tsconfig.json (hunk 9 above)
Comment (ug02fast) [reply tsconfig.json:3]: Not really sure. I hope so. It was one of the steps I took in debugging errors.

**Pull request merged**